miner/algo_default.go

```diff
@@ -0,0 +1,150 @@
+package miner
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type GreedyBuilder struct {
+	chainData   chainData
+	interrupt   *int32
+	bundleCache *BundleCache
+}
+
+func NewGreedyBuilder(chain *core.BlockChain, chainConfig *params.ChainConfig, blacklist map[common.Address]struct{}, interrupt *int32, bundleCache *BundleCache) *GreedyBuilder {
+	return &GreedyBuilder{
+		chainData:   chainData{chainConfig, chain, blacklist},
+		interrupt:   interrupt,
+		bundleCache: bundleCache,
+	}
+}
+
+func (g *GreedyBuilder) doBuildBlock(inputEnv *environment, orders []types.Order) (*environment, []types.Order) {
```

Comment:
A simple solution to avoid having non-simulated bundles be a part of the input orders is to pass the mempool and mevbundles separately into the doBuildBlock, since if we don't simulate up front there is no meaning to the orders